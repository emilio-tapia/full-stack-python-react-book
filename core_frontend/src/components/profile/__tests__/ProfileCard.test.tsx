import { describe, it, expect, beforeEach } from 'vitest';
import { render, screen, fireEvent } from '../../../utils/test.utils';
import ProfileCard from '../ProfileCard';

const userData = {
  id: '0590cd67-eacd-4299-8413-605bd547ea17',
  first_name: 'Mossie',
  last_name: 'Murphy',
  name: 'Mossie Murphy',
  post_count: 3,
  email: 'Mossie@yopmail.com',
  bio: 'Omnis necessitatibus facere vel in est provident sunt tempora earum accusantium debitis vel est architecto minima quis sint et asperiores.',
  username: 'MossieMurphy',
  avatar: null,
  created: '2022-08-19T17:31:03.310Z',
  updated: '2022-08-20T07:38:47.631Z',
};

describe('ProfileCard', () => {
  it('Render ProfileCard component', () => {
    render(<ProfileCard user={userData} />);

    const profileCard = screen.getByTestId('profile-card');
    expect(profileCard).toBeInTheDocument();

    const nameElement = screen.getByText(userData.name);
    expect(nameElement).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<ProfileCard user={userData} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
