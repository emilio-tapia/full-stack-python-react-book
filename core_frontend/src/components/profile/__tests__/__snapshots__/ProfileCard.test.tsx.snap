// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileCard > matches snapshot 1`] = `
<div
  class="border-0 p-2 card"
  data-testid="profile-card"
>
  <div
    class="d-flex "
  >
    <img
      class="my-3 border border-primary border-2 rounded-circle"
      height="48"
      width="48"
    />
    <div
      class="card-body"
    >
      <div
        class="fs-6 card-title h5"
      >
        Mossie Murphy
      </div>
      <button
        class="btn btn-primary"
        type="button"
      >
        See profile
      </button>
    </div>
  </div>
</div>
`;
